src/transfer-manager.ts

```diff
@@ -183,28 +203,52 @@ class XMLMultiPartUploadHelper implements MultiPartUploadHelper {
     };
   }
 
+  #setGoogApiClientHeaders(headers: Headers = {}): Headers {
+    let headerFound = false;
+
+    for (const [key, value] of Object.entries(headers)) {
+      if (key.toLocaleLowerCase().trim() === 'x-goog-api-client') {
+        headerFound = true;
+
+        // Prepend command feature to value, if not already there
+        if (!value.includes(GCCL_GCS_CMD_FEATURE.UPLOAD_SHARDED)) {
+          headers[
+            key
+          ] = `${value} gccl-gcs-cmd/${GCCL_GCS_CMD_FEATURE.UPLOAD_SHARDED}`;
+        }
+        break;
+      }
+    }
+
+    // If the header isn't present, add it
+    if (!headerFound) {
```

Comment:
We'd have to remember to update this line every time we add something new to headers. Is there a better way to do this?

Reply:
We can have a centralized place for this, however it would be cleaner to first complete the gaxios migration.

Reply:
This is also already matching the pattern in resumable uploads.

Reply:
here's an example in [#2296](https://github.com/googleapis/nodejs-storage/blob/3ccc9bc9b87e5b537286ea8ce78ac39acb5a1290/src/nodejs-common/util.ts#L1023).

Yeah, thats fine. Please create a TODO & ticket to follow up

Reply:
https://github.com/googleapis/nodejs-storage/issues/2314